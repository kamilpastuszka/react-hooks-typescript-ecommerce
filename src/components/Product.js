import React from "react";

export default function Product() {
  return (
    <div className="col-sm-12 col-md-5">
      <div className="card">
        <div className="card-body" />
        <div className="card-footer">
          <button className="btn btn-primary" />
        </div>
      </div>
    </div>
  );
}
