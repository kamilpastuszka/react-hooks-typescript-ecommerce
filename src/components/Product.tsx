import React from "react";

export default function Product(props) {
  return (
    <div className="col-sm-12 col-md-6 mt-5">
      <div className="card">
        <div className="card-body">{props.name}</div>
        <div className="card-footer">{props.price} </div>
        <button className="btn btn-primary" onClick={props.addProduct}>
          add
        </button>
      </div>
    </div>
  );
}
