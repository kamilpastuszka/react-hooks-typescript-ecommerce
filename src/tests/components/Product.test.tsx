import React from "react";
import { shallow } from "enzyme";
import Product from "../../components/Product";
import toJson from "enzyme-to-json";

test("should render Basket correctly", () => {
  const onAddProductSpy = jest.fn();
  const wrapper = shallow(
    <Product name={"item 1"} price={100} addProduct={onAddProductSpy} />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
