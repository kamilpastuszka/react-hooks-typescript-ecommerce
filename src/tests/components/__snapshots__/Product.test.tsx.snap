// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Basket correctly 1`] = `
<div
  className="col-sm-12 col-md-5 mt-5 justify-content-around"
>
  <div
    className="card"
  >
    <styled.div
      className="card-body py-5"
    >
      <styled.div>
        item 1
      </styled.div>
    </styled.div>
    <styled.div
      className="card-footer d-flex justify-content-around"
    >
      100
      <div>
        <button
          className="btn btn-primary"
          onClick={[MockFunction]}
        >
          add
        </button>
      </div>
    </styled.div>
  </div>
</div>
`;
